### DISCUSSION TITLE
When altering next/config at runtime, the changes to the config values do not update client side in `__NEXT_DATA__.runtimeConfig`

### DISCUSSION THREAD

**Discussion by mattruby** [vercel/next.js#24393] [Help]
### What version of Next.js are you using?

10.1.3

### What version of Node.js are you using?

15.9.0

### What browser are you using?

Chrome

### What operating system are you using?

macOS

### How are you deploying your application?

next start

### Describe the Bug

setConfig at runtime is not updating client side in `__NEXT_DATA__.runtimeConfig`.  Server side, the values are updated and usable both in normal and api pages.  

When rendering server side, the correct updated config values are shown.  However, when React renders client side, it overwrites the ssr values with the older `__NEXT_DATA__.runtimeConfig` values.  

### Expected Behavior

I would expect setConfig to update `__NEXT_DATA__.runtimeConfig` at runtime. 

### To Reproduce

https://github.com/mattruby/next-config-update-bug
First, run the development server:

```bash
npm run dev
# or
yarn dev
```

Open [http://localhost:3000](http://localhost:3000) with your browser to see the publicRuntimeConfig.

Open a tab and hit: [http://localhost:3000/api/updateConfig](http://localhost:3000/api/updateConfig). This will update the runtime next.js config.

Refresh [http://localhost:3000](http://localhost:3000), you'll see the old values

Open your console to see:
```
...
Warning: Text content did not match. Server: "{
  "publicRuntimeConfig": {
    "test": "This has been updated! 1619187877202"
  }
}" Client: "{
  "publicRuntimeConfig": {
    "test": "This is test value's value."
  }
}"
...
```
The update will show server side. But it seems that the `__NEXT_DATA__.runtimeConfig` is cached and refuses to update client side.

**Comment by mattruby** [comment]
I'm more than happy to help sort this out.  If someone could give me a hint on where to start.  I've chased runtimeConfig

**Comment by timneutkens** [comment]
`setConfig` is not a public api to be used.

**Reply by mattruby** [reply]
Yikes.  So my use case is not going to be covered by `next/config`. OK, good to know.  
And I'll stop writing that blog article about how to update `next/config` at runtime! :-) Haha!
